core/templates/dev/head/pages/learner_dashboard/LearnerDashboard.js

```diff
@@ -191,6 +227,100 @@ oppia.controller('LearnerDashboard', [
       return value;
     };
 
+    $scope.sortFeedbackFunction = function(entity) {
+      return entity[$scope.currentFeedbackThreadsSortType];
+    }
+
+    $scope.onClickThread = function(
+      explorationId, threadId, explorationTitle) {
+      var threadDataUrl = (
+        '/learnerdashboardthreadhandler/' + explorationId + '/' + threadId);
+      $scope.threadTitle = explorationTitle;
+      $scope.feedbackThreadActive = true;
+      $scope.explorationId = explorationId;
+      $scope.threadId = threadId;
+
+      for (var index = 0; index < $scope.threadSummaries.length; index++) {
+        if ($scope.threadSummaries[index].exploration_id === explorationId &&
+            $scope.threadSummaries[index].thread_id === threadId) {
+          $scope.threadIndex = index;
```

Comment:
Where's $scope.threadIndex defined? Where is it set back to null?

Also, if you don't use it in the HTML then use 'var' instead to keep it private to the controller.

This controller is getting long, btw. Try pulling out coherent subparts of it into a directive, if you can.